File: Backend/Travle.Model/Responses/OrganizerTourStatRow.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// Per-tour line of the organizer statistics screen: how each of the organizer's own tours is
    /// performing. <see cref="NetEarnings"/> is captured minus refunds minus commission on that tour;
    /// <see cref="AverageRating"/> is 0 when the tour has no (non-removed) reviews yet.
    /// </summary>
    public class OrganizerTourStatRow
    {
        public string TourName { get; set; } = string.Empty;
        public int Bookings { get; set; }
        public decimal NetEarnings { get; set; }
        public double AverageRating { get; set; }
        public int ReviewCount { get; set; }
    }
}

File: UI/travle_core/lib/src/models/organizer_tour_stat_row.dart
import 'package:json_annotation/json_annotation.dart';

part 'organizer_tour_stat_row.g.dart';

/// Per-tour line of the organizer statistics screen (mirrors the backend
/// `OrganizerTourStatRow`). [averageRating] is 0 when the tour has no reviews yet.
@JsonSerializable()
class OrganizerTourStatRow {
  OrganizerTourStatRow({
    required this.tourName,
    required this.bookings,
    required this.netEarnings,
    required this.averageRating,
    required this.reviewCount,
  });

  final String tourName;
  final int bookings;
  final double netEarnings;
  final double averageRating;
  final int reviewCount;

  factory OrganizerTourStatRow.fromJson(Map<String, dynamic> json) =>
      _$OrganizerTourStatRowFromJson(json);

  Map<String, dynamic> toJson() => _$OrganizerTourStatRowToJson(this);
}
